acceptance/analyzer_test.go

```diff
@@ -76,317 +77,494 @@ func TestAnalyzer(t *testing.T) {
 	h.DockerBuild(t,
 		analyzeImage,
 		analyzeDockerContext,
-		h.WithFlags("-f", filepath.Join(analyzeDockerContext, dockerfileName)),
+		h.WithFlags(
+			"--build-arg", fmt.Sprintf("cnb_platform_api=%s", api.Platform.Latest()),
+			"-f", filepath.Join(analyzeDockerContext, dockerfileName),
+		),
 	)
 	defer h.DockerImageRemove(t, analyzeImage)
 
-	spec.Run(t, "acceptance-analyzer", testAnalyzer, spec.Parallel(), spec.Report(report.Terminal{}))
+	for _, api := range api.Platform.Supported {
+		spec.Run(t, "acceptance-analyzer/"+api.String(), testAnalyzerBuilder(api.String()), spec.Parallel(), spec.Report(report.Terminal{}))
+	}
 }
 
-func testAnalyzer(t *testing.T, when spec.G, it spec.S) {
-	var copyDir, containerName, cacheVolume string
-
-	it.Before(func() {
-		containerName = "test-container-" + h.RandString(10)
-		var err error
-		copyDir, err = ioutil.TempDir("", "test-docker-copy-")
-		h.AssertNil(t, err)
-	})
-
-	it.After(func() {
-		if h.DockerContainerExists(t, containerName) {
-			h.Run(t, exec.Command("docker", "rm", containerName))
-		}
-		if h.DockerVolumeExists(t, cacheVolume) {
-			h.DockerVolumeRemove(t, cacheVolume)
-		}
-		os.RemoveAll(copyDir)
-	})
-
-	when("called without an app image", func() {
-		it("errors", func() {
-			cmd := exec.Command(
-				"docker", "run", "--rm",
-				"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
-				analyzeImage,
-				ctrPath(analyzerPath),
-			) // #nosec G204
-			output, err := cmd.CombinedOutput()
-
-			h.AssertNotNil(t, err)
-			expected := "failed to parse arguments: received 0 arguments, but expected 1"
-			h.AssertStringContains(t, string(output), expected)
+func testAnalyzerBuilder(apiString string) func(t *testing.T, when spec.G, it spec.S) {
```

Comment:
Why did we rename this function?

Reply:
I think it's because this function doesn't actually run "test analyzer" but rather returns the function that will.

Reply:
Ohhh. I missed this part. Thanks!